consuming-webservices/Networking/RequestHandler.swift

```diff
@@ -45,12 +45,51 @@ struct HTTPRequestHandler: RequestHandler {
     var body: Any?
     
     func execute( callback: @escaping (Result<Any>) -> Void) {
-        var returnValue = 0
-        if let body = body as? [String: String] {
-            returnValue = body["args"] == "on" ? 1 : 0
+        guard
+            let url = URL(string: path) else {
+                callback(.failure(RequestError.invalidURL))
+                return
         }
         
-        let fakeData = ["return_value": returnValue, "result": 0]
-        callback(.success(fakeData))
+        var request = URLRequest(url: url)
+        request.httpMethod = method.rawValue
+        request.allHTTPHeaderFields = headers
+        do {
```

Comment:
This `do catch` feels too generic. You may end up with more `try`s and want to handle each error appropriately.